### DISCUSSION TITLE
How to query a collection and return a random document?

### DISCUSSION THREAD

**Discussion by InfiniteNex** [GodotNuts/GodotFirebase#180] [Q&A] [answered]
I found this answer: https://stackoverflow.com/questions/46798981/firestore-how-to-get-random-documents-in-a-collection  
But I cant quite wrap my head around translating it within this module.

Also, when I query and print the result, all I get is this: [{readTime:2021-04-18T17:23:18.498566Z}]
How can I make it return the document name for example, instead of this readTime stuff?

**Comment by BearDooks** [comment, marked as answer]
@InfiniteNex Sorry it took so long to get back to you.

In terms of selecting a random document in a collection, I would need to look into that. You are honestly the first person that has asked the question.

In regards to printing the document, take a look at this part of the wiki: https://github.com/GodotNuts/GodotFirebase/wiki/Firestore#get-a-document

You can also look at this test code I have when we test a new version of the plugin


    # Connect to the collection
    _collection = Firebase.Firestore.collection('Firebasetester')
	    
    # Connect needed signals for testing
    _collection.connect("add_document", self, "on_document_add")
    _collection.connect("get_document", self, "on_document_get")
    _collection.connect("update_document", self, "on_document_update")
    _collection.connect("delete_document", self, "on_document_delete")
    _collection.connect("error", self, "on_document_error")
    
    # Get Document1

    _collection.get('Document1')
    _document = yield(_collection, "get_document")
    
    # Print Document1
    print("Trying to print contents of Document1")
    print(_document)

~ Chuck
